pydis_site/apps/api/viewsets/bot/offensive_message.py

```diff
@@ -0,0 +1,62 @@
+from rest_framework.mixins import (
+    CreateModelMixin,
+    DestroyModelMixin,
+    ListModelMixin,
+    UpdateModelMixin
+)
+from rest_framework.viewsets import GenericViewSet
+
+from pydis_site.apps.api.models.bot.offensive_message import OffensiveMessage
+from pydis_site.apps.api.serializers import OffensiveMessageSerializer
+
+
+class OffensiveMessageViewSet(
+    CreateModelMixin, ListModelMixin, DestroyModelMixin, UpdateModelMixin, GenericViewSet
```

Comment:
Is there ever a case where we use PUT or PATCH methods to update the entries in the table? I don't see it mentioned in the docstring and there are no tests for those methods in `test_offensive_message`, but you did include the [`UpdateModelMixin`](https://www.django-rest-framework.org/api-guide/generic-views/#updatemodelmixin) here, which will implement these (partial/full) update methods for existing entries.

This is related to the next comment as well, but I don't really see a use-case for updating existing entries; we mainly want to be able to add them, list them, and delete them when we're done with them.

Using these mixins:
```suggestion
    CreateModelMixin, ListModelMixin, DestroyModelMixin, GenericViewSet
```
will make sure we're able to create, list, and destroy entries, but nothing else.

All these actions should be confirmed by tests and, preferably, we should test that using a method that's not allowed (PUT/PATCH) results in a `405` response.